geotrek/api/v2/filters.py

```diff
@@ -411,6 +413,10 @@ def filter_queryset(self, request, queryset, view):
         types = request.GET.get('types')
         if types:
             qs = qs.filter(type__in=types.split(','))
+        trek = request.GET.get('trek', None)
+        if trek is not None:
+            t = Trek.objects.get(pk=trek)
```

Comment:
Use `filter` instead of `get` to avoid exceptions when bad trek ID is provided

Reply:
Done